neqo-common/src/qlog.rs

```diff
@@ -65,6 +65,19 @@ impl NeqoQlog {
         });
     }
 
+    /// If logging enabled, closure may generate an event to be logged.
+    pub fn add_event_data<F>(&mut self, f: F)
+    where
+        F: FnOnce() -> Option<qlog::events::EventData>,
```

Comment:
I can't see any places where an instance of `F` returns `None`.  So maybe you can just do:

```suggestion
        F: FnOnce() -> qlog::events::EventData,
```